pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -215,76 +215,98 @@ func chooseRandomNode(windowsVMs map[types.WindowsVM]bool) types.WindowsVM {
 	return nil
 }
 
-// deleteWindowsVMs deletes the required number of Windows VMs from the cluster and returns a bool indicating the
-// status of deletion. This method will return false if any of the VMs fail to get deleted.
+// removeWorkerNode terminates the underlying VM and removes the given vm from the list of VMs
+func (r *ReconcileWindowsMachineConfig) removeWorkerNode(vm types.WindowsVM) error {
+	// VM is missing credentials, this can occur if there was a failure initially creating it. We can consider the
+	// actual VM terminated as there is nothing we can do with it.
+	if vm.GetCredentials() == nil || len(vm.GetCredentials().GetInstanceId()) == 0 {
+		delete(r.windowsVMs, vm)
+		return nil
+	}
+
+	// Terminate the instance via its instance id
+	id := vm.GetCredentials().GetInstanceId()
+	log.V(1).Info("removing the Windows VM", "ID", id)
```

Comment:
s/removing/destroying

We destroy the VM and remove the worker node from the cluster. That is my reasoning for this usage.